tests/test_everflow_testbed.py

```diff
@@ -0,0 +1,664 @@
+                  'session_ttl' : session_info['session_ttl'],
+                  'session_dscp' : session_info['session_dscp'],
+                  'acl_stage' : acl_stage,
+                  'mirror_stage' : mirror_type,
+                  'expect_received' : expect_receive }
+        params.update(kwargs)
+        ptf_runner(host=ptfhost,
+                   testdir="acstests",
+                   platform_dir="ptftests",
+                   testname="everflow_tb_test.EverflowTest" if test_name is None else test_name,
+                   params=params,
+                   socket_recv_size=16384,
+                   log_file="/tmp/{}.{}.log".format(request.cls.__name__, request.function.__name__))
+
+    return _partial_ptf_runner
+
+def add_route(duthost, prefix, nexthop):
+    """
+    utility to add route
+    duthost: fixture that have duthost information
+    prefix:  Ip prefix 
+    nexthop: nexthop
+    return:pass
+    """
+    duthost.shell("vtysh -c 'configure terminal' -c 'ip route {} {}'".format(prefix, nexthop))
+    pass
+
+def remove_route(duthost, prefix, nexthop):
+    """
+    utility to remove route
+    duthost: fixture that have duthost information
+    prefix:  Ip prefix 
+    nexthop: nexthop
+    return:pass
+    """
+    duthost.shell("vtysh -c 'configure terminal' -c 'no ip route {} {}'".format(prefix, nexthop))
+    pass
+
+def get_neighbor_info(duthost, dest_port, resolved = True):
+
+    if resolved == False:
+        return '20.20.20.100', None
+
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+
+    for bgp_peer in mg_facts['minigraph_bgp']:
+        if bgp_peer['name'] == mg_facts['minigraph_neighbors'][dest_port]['name'] and ipaddress.IPAddress(bgp_peer['addr']).version == 4:
+            peer_ip = bgp_peer['addr']
+            break
+
+    return peer_ip, duthost.shell("ip neigh show %s | awk -F' ' '{print $5}'" %peer_ip)['stdout']
+
+
+class BaseEverflowTest(object):
+
+    @pytest.fixture(params=['tor', 'spine'])
+    def dest_port_type(self, request):
+        """
+        used to parametrized test cases on dest port type
+        :param request: pytest request object
+        :return: destination port type
+        """
+        return request.param
+
+    @pytest.fixture(scope='class')
+    def setup_mirror_session(self, duthost):
+        """Setup the Everflow Session"""
+
+        session_name =  "test_session_1"
+        session_src_ip =  "1.1.1.1"
+        session_dst_ip = "2.2.2.2"
+        session_ttl = "1"
+        session_dscp = "8"
+
+        session_prefixlens = ["24", "32"]
+        session_prefixes = []
+        for prefixlen in session_prefixlens:
+            session_prefixes.append(str(ipaddress.IPNetwork(session_dst_ip + "/" + prefixlen).network) + "/" + prefixlen)
+
+        duthost.command('config mirror_session add {} {} {} {} {}'.format(session_name, session_src_ip, session_dst_ip, session_dscp, session_ttl))
```

Comment:
Can you make these format strings consistent throughout the test? I noticed in a couple other places you used the `%` format (e.g. `"words words words %s" % words"`). I would personally stick to using `.format()` throughout, but as long as it's consistent you can use whichever one you feel is more clear. :)

Reply:
@daall Thanks. Updated